backend/internal/server/apis/schedule/pkg/analysis/analysis.go

```diff
@@ -0,0 +1,142 @@
+package analysis
+
+import (
+	"fmt"
+	"github.com/jak103/powerplay/internal/server/apis/schedule/pkg/models"
+	"github.com/jak103/powerplay/internal/utils/log"
+
+	"sort"
+	"time"
+)
+
+func RunTimeAnalysis(games []models.Game) (models.SeasonStats, map[string]models.TeamStats) {
+	seasonStats := models.SeasonStats{
+		TotalGames: len(games),
+	}
+	teamStats := make(map[string]models.TeamStats)
+
+	for _, game := range games {
+		var team1Stats models.TeamStats
+		var team2Stats models.TeamStats
+		var ok bool
+
+		if team1Stats, ok = teamStats[game.Team1Name]; !ok {
+			team1Stats = newStats(game.League, game.Team1Name)
+		}
+
+		if team2Stats, ok = teamStats[game.Team2Name]; !ok {
+			team2Stats = newStats(game.League, game.Team2Name)
+		}
+
+		team1Stats.Games = append(team1Stats.Games, game)
+		team2Stats.Games = append(team2Stats.Games, game)
+
+		earlyLateGames(game, &seasonStats, &team1Stats, &team2Stats)
+
+		daysOfTheWeek(game, &team1Stats, &team2Stats)
+
+		teamStats[game.Team1Name] = team1Stats
+		teamStats[game.Team2Name] = team2Stats
+	}
+
+	seasonEarlyHigh := int(seasonStats.EarlyPercentage()*10.0) + 1 // TODO took a shortcut here and just hardcoded the 10 games
+
+	for _, team := range teamStats {
+		team.Balanced = team.EarlyGames <= seasonEarlyHigh
+		teamStats[team.Name] = team
+	}
+
+	timeBetweenGames(teamStats)
+
+	printStats(seasonStats, teamStats)
+
+	return seasonStats, teamStats
+}
+
+func newStats(league, team string) models.TeamStats {
+	return models.TeamStats{
+		League:        league,
+		Name:          team,
+		DaysOfTheWeek: make(map[time.Weekday]int),
+		Games:         make([]models.Game, 0),
+	}
+}
+
+func earlyLateGames(game models.Game, season *models.SeasonStats, team1, team2 *models.TeamStats) {
+	if game.IsEarly {
+		team1.EarlyGames += 1
+		team2.EarlyGames += 1
+		season.EarlyGames += 1
+	} else {
+		team1.LateGames += 1
+		team2.LateGames += 1
+		season.LateGames += 1
+	}
+}
+
+func daysOfTheWeek(game models.Game, team1, team2 *models.TeamStats) {
+	team1.DaysOfTheWeek[game.Start.Weekday()] += 1
+	team2.DaysOfTheWeek[game.Start.Weekday()] += 1
+}
+
+func timeBetweenGames(teamStats map[string]models.TeamStats) {
+	for team := range teamStats {
+		stats := teamStats[team]
+		for i := 1; i < len(stats.Games); i += 1 {
+			previousGame := stats.Games[i-1]
+			currentGame := stats.Games[i]
+
+			betweenDuration := currentGame.Start.Sub(previousGame.Start)
+
+			days := int(betweenDuration.Hours() / 24)
+			stats.DaysBetweenGames = append(stats.DaysBetweenGames, days)
+
+			stats.AverageDaysBetweenGames += float32(days)
+		}
+
+		stats.AverageDaysBetweenGames /= float32(len(stats.DaysBetweenGames))
+
+		teamStats[team] = stats
+	}
+}
+
+func printStats(seasonStats models.SeasonStats, teamStats map[string]models.TeamStats) {
+	log.Info("Early games: %v/%v (%v%%)\n", seasonStats.EarlyGames, seasonStats.TotalGames, (float32(seasonStats.EarlyGames)/float32(seasonStats.TotalGames))*100)
```

Comment:
For the analysis do we want it printed out to the logs? It seems like we would want it output separately from the rest of the logs, where it is potentially interwoven with other log statements like user requests

Reply:
Do you have any thoughts on how to separate it? Maybe we return the analysis logs to the user or save them in the db attached to the schedule model?

Reply:
For the most part, I wonder if we can just get rid of printStats. I think it was just a way for him to see how well his scheduling algorithm did, but I am not 100%

Reply:
How does everyone feel about changing the log level to debug for now then if we want them later we can do something else with it?